cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1087,16 +1100,14 @@ void snappy_decompress(device_span<gpu_inflate_input_s> comp_in,
   CUDF_EXPECTS(nvcomp_status == nvcompStatus_t::nvcompSuccess,
                "unable to perform snappy decompression");
 
-  CUDF_EXPECTS(thrust::equal(rmm::exec_policy(stream),
-                             uncompressed_data_sizes.begin(),
-                             uncompressed_data_sizes.end(),
-                             actual_uncompressed_data_sizes.begin()),
-               "Mismatch in expected and actual decompressed size during snappy decompression");
-  CUDF_EXPECTS(thrust::equal(rmm::exec_policy(stream),
-                             statuses.begin(),
-                             statuses.end(),
-                             thrust::make_constant_iterator(nvcompStatus_t::nvcompSuccess)),
-               "Error during snappy decompression");
+  const bool check_for_errors = any_page_failure != nullptr;
+  if (check_for_errors) {
+    dim3 block(128);
+    dim3 grid((num_comp_pages + block.x - 1) / block.x);
+    nvcomp_snappy_decompress_check<<<grid, block, 0, stream.value()>>>(statuses.data(),
```

Comment:
Rather than have a duplicate check for cuIO internal and nvcomp paths, you can use a thrust transform to write the status values to `comp_stat` and check it later in the `decompress_page_data` function below. That way, we get checking for all formats at the same time

Reply:
Great suggestion, I'll fix that

Reply:
I fixed this, however I used a custom kernel to convert the nvcomp status rather than thrust::for_each_n since local testing showed this to be 2-3s faster for TPC-DS query 9.